### PR TITLE
Remove the proposals workaround from UTXO conformance, add LEDGER conformance

### LINKED ISSUE #4672: LEDGER conformance

### PR DESCRIPTION
**Pull request by Soupstraw**
# Description

This PR removes a workaround in the UTXO conformance test. I modified the spec so that it doesn't add new deposits in the UTXO rule. I also moved the UTXO conformance logic into dedicated modules. 

I had to disable some certificates, because the Generic generators aren't generating the deposits field correctly.

I also added conformance testing for the LEDGER rule. It seems to be succeeding in most cases, but sometimes the generators generates values that make the test fail.

closes #4672

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages.
      **_New section is never added with the code changes._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated.
      **_If you change the bounds in a cabal file, that package itself must have a version increase._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/AdaPots.hs [hunk 1]
```diff
@@ -153,7 +154,9 @@ consumedTxBody ::
   Consumed
 consumedTxBody txBody pp dpstate utxo =
   Consumed
-    { conInputs = coinBalance (txInsFilter utxo (txBody ^. inputsTxBodyL))
+    { conInputs =
+        traceShow ("utxo: " <> show utxo <> "\n\ninputs: " <> show (txBody ^. inputsTxBodyL)) $
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/AdaPots.hs]: Gotta be careful about retaining debugging traces
```suggestion
```

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs [hunk 2]
```diff
@@ -98,6 +100,16 @@ deriving instance Eq (PParams era) => Eq (LedgerEnv era)
 instance NFData (PParams era) => NFData (LedgerEnv era) where
   rnf (LedgerEnv _slotNo _ix pp _account _mempool) = rnf pp
 
+instance EraPParams era => EncCBOR (LedgerEnv era) where
+  encCBOR LedgerEnv {..} =
```
Comment (lehins) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs]: I suggest adding pattern match here, so we don't forget to change this instance when or if `LedgerEnv` changes:
```suggestion
  encCBOR env@LedgerEnv {..} =
    let LedgerEnv _ _ _ _ _ = env in
```

**Review by lehins — DISMISSED** [review]
We need to get rid of `traceShow`. Other than that it looks good

**Review by williamdemeo — APPROVED** [review]
I won't pretend to have carefully checked and understood everything here, but what I do understand looks good.  I ran the tests (`nix develop` and `cabal test all`) and all that are implemented passed.

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs (hunk 2 above)
Comment (Lucsanszky) [reply eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs]: This triggers the `unused-pattern-binds` warning unfortunately.

Comment (teodanciu) [reply eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs]: You could do this instead, without warnings:
```suggestion
  encCBOR env@(LedgerEnv _ _ _ _ _) =
    let LedgerEnv {..} = env in
```

Comment (lehins) [reply eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledger.hs]: Thank you @teodanciu I forgot the order in which we usually do this :smile:

**Pull request merged**